dashboard/app/models/pd/accepted_program.rb

```diff
@@ -20,4 +20,10 @@ class Pd::AcceptedProgram < ActiveRecord::Base
   validates_inclusion_of :course, in: Pd::TeacherApplication::PROGRAM_DETAILS_BY_COURSE.keys, if: -> {course.present?}
   validates_presence_of :user
   validates_uniqueness_of :user_id, scope: :course, message: 'already has an entry for this course'
+
+  TEACHER_CONS_CITY_REGEX = /Houston|Phoenix|Philadelphia/
+
+  def teachercon?
+    !!(workshop_name =~ TEACHER_CONS_CITY_REGEX)
```

Comment:
Good call. We're doing it two different ways now, using specific strings in the [email scripts](https://github.com/code-dot-org/code-dot-org/blob/staging/bin/mail/2017-teacher-application-responses/teacher_application_decision_processor.rb#L48-L52) and this regex in the [program registration form](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/pd-program-registration/index.haml#L8). I'll use the specific strings and put it in one place in shared spot that both code paths can leverage. 

No, we don't have a teachercon bit on the workshop model, but it doesn't belong there anyway because teachercons don't have an associated workshop entry. Registration happens on the program registration form.